### PR TITLE
fix(inapp): remove implemented todos and add clearMessages API

### PR DESCRIPTION
**Pull request by Samaritan1011001**
#### Description of changes

Removes stale TODOs that already have implementations and add the missing clearMessages APIs

#### Description of how you validated changes
yarn test and test cases

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/auth/__tests__/providers/cognito/identityIdProvider.test.ts [hunk 1]
```diff
@@ -4,8 +4,6 @@
 import { authAPITestParams } from './testUtils/authApiTestParams';
 import { Amplify, Identity, ResourcesConfig } from '@aws-amplify/core';
 import { DefaultIdentityIdStore } from '../../../src/providers/cognito/credentialsProvider/IdentityIdStore';
-
-// TODO(V6): import these from top level core/ and not lib/
```
Comment (cshfang) [inline packages/auth/__tests__/providers/cognito/identityIdProvider.test.ts:8]: Why was this removed without change though :/

File: packages/auth/src/providers/cognito/credentialsProvider/IdentityIdStore.ts [hunk 2]
```diff
@@ -34,7 +34,6 @@ export class DefaultIdentityIdStore implements IdentityIdStore {
 
 	async loadIdentityId(): Promise<Identity | null> {
 		assertIdentityPoolIdConfig(this.authConfig?.Cognito);
-		// TODO(v6): migration logic should be here
```
Comment (cshfang) [inline packages/auth/src/providers/cognito/credentialsProvider/IdentityIdStore.ts:37]: Ditto this

File: packages/notifications/src/inAppMessaging/providers/pinpoint/apis/clearMessages.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import {
+	InAppMessagingValidationErrorCode,
+	assertServiceError,
+} from '../../../errors';
+import { assertIsInitialized } from '../../../utils';
+
+/**
+ * Clear locally cached messages.
+ *
+ * @throws service exceptions - Thrown when the underlying Pinpoint service returns an error.
```
Comment (cshfang) [inline packages/notifications/src/inAppMessaging/providers/pinpoint/apis/clearMessages.ts:16]: Would this API ever return a service exception?

File: packages/auth/__tests__/providers/cognito/identityIdProvider.test.ts (hunk 1 above)
Comment (Samaritan1011001) [reply packages/auth/__tests__/providers/cognito/identityIdProvider.test.ts]: thank you. Intended to remove only the other one, this one needs more work and will come in the next PR so reverted it.

File: packages/notifications/src/inAppMessaging/providers/pinpoint/apis/clearMessages.ts (hunk 3 above)
Comment (Samaritan1011001) [reply packages/notifications/src/inAppMessaging/providers/pinpoint/apis/clearMessages.ts]: no, removed

**Review by jimblanc — DISMISSED** [review]

**Review by cshfang — DISMISSED** [review]

**Review by cshfang — APPROVED** [review]
